src/commandline_options.cpp

```diff
@@ -558,28 +556,33 @@ commandline_options::commandline_options(const std::vector<std::string>& args)
 
 void commandline_options::parse_log_domains_(const std::string &domains_string, const int severity)
 {
-	const std::vector<std::string> domains = utils::split(domains_string, ',');
-	for (const std::string& domain : domains)
+	if(std::vector<std::string> domains = utils::split(domains_string, ','); !domains.empty())
 	{
-		if (!log)
-			log = std::vector<std::pair<int, std::string>>();
-		log->emplace_back(severity, domain);
+		if(!log)
+		{
+			log.emplace();
+		}
+		for (auto&& domain : domains)
+		{
+			assert(log && "`log` must have been initialized before this loop");
+			log->emplace_back(severity, std::move(domain));
+		}
 	}
 }
 
-void commandline_options::parse_log_strictness (const std::string & severity ) {
-	static const std::array<lg::logger const*, 4> loggers {{&lg::err(), &lg::warn(), &lg::info(), &lg::debug()}};
+void commandline_options::parse_log_strictness (const std::string & severity) {
+	static const lg::logger* const loggers[4] {&lg::err(), &lg::warn(), &lg::info(), &lg::debug()};
```

Comment:
want to point my reasons out

We're not really using anything from what std array offers over a C style array, in this case

The code using array is more complex (as array is an aggregate and contains an aggregate so the double {})

Also, there's no need for its copy semantics or any other of its advantages (like the member side or others). Don't get me wrong, I'm all for std array but only when it brings something to the table.

Otherwise it's a cost we pay in compilation time for little to no gain.

All that being said I respect your opinion and will change it if you desire